test/unit/util.go

```diff
@@ -0,0 +1,205 @@
+package unit
+
+import (
+	v1 "github.com/openshift/api/operator/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type routeBuilder struct {
+	name          types.NamespacedName
+	labels        map[string]string
+	admittedICs   []string
+	unAdmittedICs []string
+}
+
+func NewRouteBuilder() *routeBuilder {
+	return &routeBuilder{
+		name: types.NamespacedName{Name: "sample", Namespace: "openshift-ingress"},
+	}
+}
+
+func (b *routeBuilder) WithName(name types.NamespacedName) *routeBuilder {
+	b.name = name
+	return b
+}
+
+func (b *routeBuilder) WithLabels(labels map[string]string) *routeBuilder {
+	b.labels = labels
+	return b
+}
+
+func (b *routeBuilder) WithAdmittedICs(admittedICs ...string) *routeBuilder {
+	b.admittedICs = admittedICs
+	return b
+}
+
+func (b *routeBuilder) WithUnAdmittedICs(unAdmittedICs ...string) *routeBuilder {
+	b.unAdmittedICs = unAdmittedICs
+	return b
+}
+
+func (b routeBuilder) Build() *routev1.Route {
+	route := &routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      b.name.Name,
+			Namespace: b.name.Namespace,
+			Labels:    b.labels,
+		},
+		Spec:   routev1.RouteSpec{},
+		Status: routev1.RouteStatus{},
+	}
+
+	if len(b.admittedICs) != 0 {
+		for _, ic := range b.admittedICs {
```

Comment:
`range` would not be executed if the input slice is empty, so the check for the length is superfluous.

Reply:
Good point.